### DISCUSSION TITLE
How to load a <FormAutocomplete> component pre-filled with multiple items?

### DISCUSSION THREAD

**Discussion by calionauta** [yesvelte/yesvelte#504] [Q&A] [answered]
I am currently using this component and it works for allowing multiple items, but I am unable to make it load with some pre-filled items using complex items.

Is it an error on my part or an issue with the component?

**Comment by TheHadiAhmadi** [comment]
@renatocaliari do you mean something like this?

```svelte
<Autocomplete multiple items={someItems} bind:value={nonEmptyArray} />
```

above code works but below code doesn't work (tomorrow I'll work on fixing this bug)

```svelte
<Autocomplete multiple key="id" items={someItems} bind:value={nonEmptyArrayOfIds} />
```

Here is a REPL: https://svelte.dev/repl/b59ecee6dc1641d29794d98aff076238?version=3.55.1

if this is not your problem, please provide more information or create an example repl.

thanks

**Reply by calionauta** [reply]
Thanks for the answer and examples in REPL.

I created my scenario in the REPLT and it didn't work. 

I think the problem could be the use of the $store. 

You can see that the store is being updated correctly, but the component is not showing the pre-filled items: https://svelte.dev/repl/3b891e7159af4a2d9e7e0137773455bc?version=3.55.1

**Reply by TheHadiAhmadi** [reply]
@renatocaliari problem is that you are creating new object for store values. You should pass items[2] and items[3] to the store. 

```js
const a = {id: "abc"}
const b = {id: "abc"}

console.log(a === b) // always false
```

```js
const obj = {id: "abc"}
const a = obj
const b = obj

console.log(a === b) // true 
```

**Reply by calionauta** [reply]
You're right. I understood the question in the example I created. Thank you for letting me know. 

But what I'm still unsure about is that in my system what is loaded into the store is what was saved in 'bind:value' with the items of the original object. 
That is, the stored object would be passed 'by reference', or not?

After submitting a form (with use:enhance), the FormAutoComplete loses these selected items.

It's possible I'm missing something. 🙄

**Reply by TheHadiAhmadi** [reply, marked as answer]
I am not sure about that. I didn't tested above scenario.

I have fixed the (multiple + key) bug.

I suggest that instead of objects, work with number/string using `key` prop.

Here is an example: https://www.yesvelte.com/docs/autocomplete#Custom_Key_(multiple)
and Here is updated REPL: https://svelte.dev/repl/135844526e6d454a82bc913058049d17?version=3.55.1

**Reply by calionauta** [reply]
Using multiple + key worked. Thanks!
